### DISCUSSION TITLE
is it possible to automatically add `width` and `height` attributes?

### DISCUSSION THREAD

**Discussion by stefanprobst** [nuxt/image#1076] [Q&A]
is it possible to automatically add `width` and `height` attributes to the image element?

for example the next.js image component [is able to do that for static image imports](https://nextjs.org/docs/app/building-your-application/optimizing/images#local-images).

**Comment by yudin-s** [comment]
Not generally, because Nuxt Image cannot know the intrinsic dimensions of every possible image source at compile time.

Next can do this for static imports because the imported local file is part of the build graph, so the build step can inspect the image and attach width/height metadata. With Nuxt Image, many common sources are remote URLs, CMS assets, provider URLs, or runtime strings, so there may be nothing available at build time to inspect.

The usual approaches are:

- pass `width` and `height` explicitly when you know them
- store dimensions in your CMS/content metadata and bind them into `<NuxtImg>`
- use fixed aspect-ratio containers in CSS when exact intrinsic size is not available
- for local assets, build your own small metadata step if you need automatic dimensions consistently

For layout stability, the important part is not necessarily automatic attributes; it is making sure the browser knows the intended aspect ratio before the image loads.

**Comment by djmtype** [comment]
@yudin-s Thanks for the insight. As I was reviewing the production build, I was shocked since Nuxt typically brings its own magic.

This one really hurts because if you're using Nuxt Studio, editors are not going to define width and height, or figure out aspect ratio for that matter. 

Even if editors were trained and remembered, this is an extra step their former CMS once provided automatically.
